src/state_circuit/memory.rs

```diff
@@ -200,46 +219,39 @@ impl<F: FieldExt, const NUM_STEPS: usize> Config<F, NUM_STEPS> {
                     }
                 }
 
+                // FIXME
+                // dummy:
+                let value = ops[0].steps[0]
+                    .as_ref()
+                    .map(|read_write| read_write.value().0);
+                region.assign_advice(
+                    || "value",
+                    self.value,
+                    31,
+                    || value.ok_or(Error::SynthesisError),
+                )?;
+
                 Ok(())
             },
         )?;
 
         Ok(bus_mappings)
     }
 
-    /// Initialise first row for a new operation.
-    fn init(
-        &self,
-        region: &mut Region<'_, F>,
-        offset: usize,
-        address: MemoryAddress<F>,
-    ) -> Result<(), Error> {
-        self.q_memory.enable(region, offset)?;
-
-        // Assign `address`
-        region.assign_advice(|| "init address", self.address, offset, || Ok(address.0))?;
-
-        // Assign `step`
-        region.assign_advice(|| "init step", self.step, offset, || Ok(F::zero()))?;
-
-        // Assign `value`
-        region.assign_advice(|| "init value", self.value, offset, || Ok(F::zero()))?;
-
-        // Assign memory_flag
-        region.assign_advice(|| "init memory", self.flag, offset, || Ok(F::one()))?;
-
-        Ok(())
-    }
-
     /// Assign cells for each step in an operation.
     fn step(
         &self,
+        step_offset: usize,
```

Comment:
If we only use the `q_memory` selector, and downgrade `q_memory_init` and `q_same_value` to synthetic selectors inside the gate, we no longer have to switch on `step_offset` here.